### PR TITLE
Add basename argument to audeer.list_*_names()

### LINKED ISSUE #37: Option to return relative paths
It would be nice to have an argument in `list_dir_names()` and `list_file_names()` that returns relative paths.

### PR DESCRIPTION
**Pull request by hagenw**
Closes #37 

Just realized I commited changes to both functions and not only to `audeer.list_dir_names()`.
So I changed the title as it should be fine to add both in a single commit.

### CODE REVIEW

**Comment by hagenw** [comment]
Should we call the argument `basenames` instead `basename` as we are also talking about `names` in the function names?

File: audeer/core/io.py [hunk 1]
```diff
@@ -276,45 +276,51 @@ def file_extension(
 
 def list_dir_names(
         path: typing.Union[str, bytes],
+        basename: bool = False,
```
Comment (frankenjoe) [inline audeer/core/io.py:279]: I suggest to make `basename` a key-word only argument like in `list_file_names`

**Comment by frankenjoe** [comment]
> Should we call the argument `basenames` instead `basename` as we are also talking about `names` in the function names?

yes, I think that makes sense

File: audeer/core/io.py (hunk 1 above)
Comment (hagenw) [reply audeer/core/io.py:280]: Good catch, that was of course my plan.

I changed it accordingly.

**Comment by hagenw** [comment]
> > Should we call the argument `basenames` instead `basename` as we are also talking about `names` in the function names?
> 
> yes, I think that makes sense

OK, I renamed the argument to `basenames`

File: audeer/core/io.py [hunk 2]
```diff
@@ -276,45 +276,52 @@ def file_extension(
...
-        >>> os.path.basename(dirs[0])
-        'path2'
+        >>> list_dir_names('path1', basenames=True)
+        ['path2']
 
     """
     path = safe_path(path)
     paths = [os.path.join(path, p) for p in os.listdir(path)]
-    return sorted([p for p in paths if os.path.isdir(p)])
+    paths = sorted([p for p in paths if os.path.isdir(p)])
```
Comment (frankenjoe) [inline audeer/core/io.py:299]: It doesn't make a difference, but I think I would sort after the if clause.

Comment (hagenw) [reply audeer/core/io.py]: Yes, that's cleaner. I moved to the very end in both functions.

**Pull request merged**